tests/monitoring_test.go

```diff
@@ -0,0 +1,128 @@
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	cdiv1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+	"kubevirt.io/containerized-data-importer/tests/framework"
+	"kubevirt.io/containerized-data-importer/tests/utils"
+)
+
+var _ = Describe("Monitoring Tests", func() {
+	fw := framework.NewFramework("monitoring-test")
+
+	BeforeEach(func() {
+		if !fw.IsPrometheusAvailable() {
+			Skip("This test depends on prometheus infra being available")
+		}
+	})
+
+	Context("Alerts", func() {
+		It("should throw CDIDataVolumeUnusualRestartCount alert when a DataVolume object restarts more than three times", func() {
+			By("creating a DataVolume")
+			dataVolume := utils.NewDataVolumeWithHTTPImport(dataVolumeName, "500Mi", validURL)
+			dataVolume, err := utils.CreateDataVolumeFromDefinition(fw.CdiClient, fw.Namespace.Name, dataVolume)
+			Expect(err).ToNot(HaveOccurred())
+			fw.ForceBindPvcIfDvIsWaitForFirstConsumer(dataVolume)
+
+			By("waiting for the DataVolume to be restarted more than three times")
+			Eventually(func() bool {
+				dv, err := fw.CdiClient.CdiV1beta1().DataVolumes(fw.Namespace.Name).Get(context.TODO(), dataVolume.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return dv.Status.RestartCount >= 4
+			})
+
+			By("checking that the CDIDataVolumeUnusualRestartCount alert is firing")
+			checkAlertExists(fw, "CDIDataVolumeUnusualRestartCount")
+
+			By("deleting the DataVolume")
+			err = utils.DeleteDataVolume(fw.CdiClient, fw.Namespace.Name, dataVolume.Name)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("should throw CDIStorageProfilesIncomplete alert if there is a incomplete StorageProfile", func() {
+			By("creating a StorageProfile with an incomplete configuration")
+			sp := &cdiv1.StorageProfile{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-storage-profile-" + rand.String(8),
+				},
+			}
+			_, err := fw.CdiClient.CdiV1beta1().StorageProfiles().Create(context.TODO(), sp, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("checking that the CDIStorageProfilesIncomplete alert is firing")
+			checkAlertExists(fw, "CDIStorageProfilesIncomplete")
+		})
+
+		It("should throw CDINotReady if CDI CR is not ready for use", func() {
+			By("creating a CDI with wrong config")
+			cdi, err := fw.CdiClient.CdiV1beta1().CDIs().Get(context.TODO(), "cdi", metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			deleteCDI(fw)
+
+			oldNodeSelector := cdi.Spec.Infra.NodeSelector
+			cdi.ResourceVersion = ""
+			cdi.UID = ""
+			cdi.Spec.Infra.NodeSelector = map[string]string{"wrong": "wrong"}
+			_, err = fw.CdiClient.CdiV1beta1().CDIs().Create(context.TODO(), cdi, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("checking that the CDINotReady alert is firing")
+			checkAlertExists(fw, "CDINotReady")
+
+			By("restoring old CDI config")
+			deleteCDI(fw)
```

Comment:
We have a special lane for destructive tests where stuff like this should run (mostly in operator_test.go),
where we take special precautions in the AfterEach of each method to ensure CDI stabilized following delete.
Might make sense to reuse some functions from that suite so a failure in this test does not crash everything that follows

Reply:
updated, moved those methods to an utils file